### PR TITLE
Add remote (nested) healthchecks

### PR DESCRIPTION
**Pull request by mikedingjan**

### CODE REVIEW

File: src/django_healthchecks/checker.py [hunk 1]
```diff
@@ -57,7 +60,9 @@ def _get_check_functions(name=None, request=None):
         if name and name != service:
             continue
 
-        if callable(func_string):
+        if func_string.startswith(('https://', 'http://')):
+            check_func = _http_healthcheck_func(func_string)
+        elif callable(func_string):
```
Comment (mvantellingen) [inline src/django_healthchecks/checker.py]: If func_string is a callable then the startswith attribute isn't available existing

File: src/django_healthchecks/views.py [hunk 2]
```diff
@@ -31,21 +31,35 @@ def get(self, request, *args, **kwargs):
...
+                break
+
         if result is None:
             raise Http404()
 
         if result in (True, False):
             status_code = 200 if result else _get_err_status_code()
             return HttpResponse(str(result).lower(), status=status_code)
 
+        if type(result) != str:
```
Comment (mvantellingen) [inline src/django_healthchecks/views.py]: ```if not isinstance(result, six.text_type)```

File: sandbox/settings.py [hunk 3]
```diff
@@ -124,9 +124,11 @@
 HEALTH_CHECKS = {
     'public': 'django_healthchecks.contrib.check_dummy_true',
     'private': 'django_healthchecks.contrib.check_dummy_false',
+    'remote': 'https://lfd-tst.labd.io/api/healthchecks/',
+    'callable': lambda: True,
 }
 
 HEALTH_CHECKS_BASIC_AUTH = {
-    '*': [('admin', 'pass')],
+    # '*': [('admin', 'pass')],
```
Comment (mvantellingen) [inline sandbox/settings.py]: ?

File: src/django_healthchecks/views.py [hunk 4]
```diff
@@ -31,21 +33,35 @@ def get(self, request, *args, **kwargs):
...
+            result = result.get(nested, None)
+            if result is None:
+                break
+
         if result is None:
             raise Http404()
 
         if result in (True, False):
             status_code = 200 if result else _get_err_status_code()
             return HttpResponse(str(result).lower(), status=status_code)
```
Comment (vdboor) [inline src/django_healthchecks/views.py]: Why is the proxied result lowercased here?

File: src/django_healthchecks/checker.py [hunk 5]
```diff
@@ -73,6 +78,24 @@ def _get_registered_health_checks():
     return getattr(settings, 'HEALTH_CHECKS', {})
 
 
+def _http_healthcheck_func(url):
+    def handle_remote_request():
+        try:
+            response = requests.get(url, timeout=_get_http_healthcheck_timeout())
+        except requests.exceptions.RequestException:
+            return False
```
Comment (vdboor) [inline src/django_healthchecks/checker.py]: check for socket errors / SSL errors? (`except IOError` catches everything on Python 3, but less on Python 2)

**Pull request merged**